### DISCUSSION TITLE
Addressing Deprecation Warnings in Next.js 14.2.4

### DISCUSSION THREAD

**Discussion by ajaymahadeven** [vercel/next.js#66872] [App Router]
### Deprecation Warnings:

- **@humanwhocodes/config-array@0.11.14**: Use @eslint/config-array instead

- **rimraf@3.0.2**: Rimraf versions prior to v4 are no longer supported

- **@humanwhocodes/object-schema@2.0.3**: Use @eslint/object-schema instead

- **glob@7.2.3**: Glob versions prior to v9 are no longer supported

Firstly, thank you for the work you've done on this framework.

When setting up a new project with Next.js 14.2.4, I encounter several deprecation warnings related to specific npm packages. These warnings can be concerning, especially for developers who prioritize maintaining a clean and efficient codebase.

**Comment by Bob-Kranson-PRFT** [comment]
Secondly, thank you for the work on this framework. I do like what it addresses regarding full stack React development.

As additional information, I found these warning also occur in Node 20.14 and Node 22.3.0. It appears to require re-integration of ESLint using ESLint 9 instead of ESLint 8.  I attempted using the NPM package.json "overrides" functionality to force glob:10 and newer rimraf and newer ESLint but that caused hangs and various additional issues/errors during build.

Package.json dependencies/devDependencies
  "dependencies": {
    "react": "latest",
    "react-dom": "latest",
    "next": "^14",
    "next-auth": "beta",
    "oracledb": "^6.5",
    "sharp": "latest"
  },
  "engines": {
    "node": "22.3.0",
    "npm": "10.8.1"
  },
  "devDependencies": {
    "@playwright/test": "latest",
    "@testing-library/jest-dom": "latest",
    "@testing-library/react": "latest",
    "@types/node": "latest",
    "@types/jest": "latest",
    "@types/react": "latest",
    "@types/react-dom": "latest",
    "@typescript-eslint/eslint-plugin": "latest",
    "@typescript-eslint/parser": "latest",
    "browserslist": "latest",
    "eslint": "^8",
    "eslint-config-next": "latest",
    "eslint-config-prettier": "latest",
    "eslint-plugin-next": "latest",
    "eslint-plugin-prettier": "latest",
    "eslint-plugin-react": "latest",
    "jest": "latest",
    "jest-environment-jsdom": "latest",
    "playwright": "latest",
    "prettier": "latest",
    "run-script-os": "latest",
    "ts-jest": "latest",
    "ts-node": "latest",
    "typescript": "latest"
  },


Adding my relevant log output(s):
Installing dependencies ...
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
npm warn deprecated @humanwhocodes/config-array@0.11.14: Use @eslint/config-array instead
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated domexception@4.0.0: Use your platform's native DOMException instead
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported

> next14-node22-ui@0.1.0 lint
> next lint
=============
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.
You may find that it works just fine, or you may not.
SUPPORTED TYPESCRIPT VERSIONS: >=4.7.4 <5.5.0
YOUR TYPESCRIPT VERSION: 5.5.2
Please only submit bug reports when using the officially supported version.
=============
#

**Reply by Bob-Kranson-PRFT** [reply]
Removing the eslint declaration from devDependencies and instead adding overrides like below will remove most of the warning though I've found multiple side effects and therefore give back and not override. This will remove warnings but causes other issues. Overrides are always use at your own risk...

  "overrides": {
    "eslint": "^9",
    "glob": "^10"
  },

$ npx reinstall

Removing node_modules directory ...
Cleaning NPM cache ..
npm warn using --force Recommended protections disabled.
Installing dependencies ...
npm warn deprecated domexception@4.0.0: Use your platform's native DOMException instead
npm warn deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead

Done!
first side effect as shown by run build:

  - Experiments (use with caution):
    · instrumentationHook

   Creating an optimized production build ...
(node:5630) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 ✓ Compiled successfully
   Linting and checking validity of types  .. ⨯ ESLint: Invalid Options: - Unknown options: useEslintrc, extensions - 'extensions' has been removed.
 ✓ Linting and checking validity of types
